libshaderc_util/src/compiler.cc

```diff
@@ -67,6 +67,45 @@ std::pair<int, string_piece> DecodeLineDirective(string_piece directive) {
   return std::make_pair(line, directive);
 }
 
+// Writes the message contained in the diagnostic parameter to *dest. Assumes
+// the diagnostic message is reported for for a binary location.
+void OutputSpvToolsDiagnostic(spv_diagnostic diagnostic, std::string* dest) {
+  assert(!diagnostic->isTextSource);
+
+  std::ostringstream os;
+  os << diagnostic->position.index << ": " << diagnostic->error;
+  *dest = os.str();
+}
+
+// Disassembles the given binary. Returns SPV_SUCCESS and writes the
+// disassembled text to *text_or_error if successful. Otherwise, writes the
+// error message to *text_or_error.
+spv_result_t DisassembleBinary(const std::vector<uint32_t>& binary,
+                               std::string* text_or_error) {
+  auto spvtools_context = spvContextCreate();
+  spv_text disassembled_text = nullptr;
+  spv_diagnostic spvtools_diagnostic = nullptr;
+
+  spv_result_t result = spvBinaryToText(
+      spvtools_context, binary.data(), binary.size(),
+      /* options = */ 0u, &disassembled_text, &spvtools_diagnostic);
+  if (result == SPV_SUCCESS) {
+    text_or_error->assign(disassembled_text->str, disassembled_text->length);
+  } else {
+    OutputSpvToolsDiagnostic(spvtools_diagnostic, text_or_error);
+  }
+
+  if (spvtools_diagnostic != nullptr) {
+    spvDiagnosticDestroy(spvtools_diagnostic);
```

Comment:
This should be safe to call on nullptr.
I now see that that behviour is neither advertised nor tested.  :-(